apps/contribution/test/contribution.js

```diff
@@ -0,0 +1,349 @@
+      let contributionObject = await contribution.getContribution(contributionCountAfter.toNumber());
+      // eslint-disable-next-line no-undef
+      assert.equal(contributionObject[1], contributorId, "contribution added belong to contributor id");
+      let isExist = await contribution.exists(contributionCountAfter.toNumber());
+      // eslint-disable-next-line no-undef
+      assert.equal(isExist, true, "contribution exist");
+    });
+  });
+
+  describe("Veto contribution", async () => {
+    it("should revert when veto from address that does not have permission", async () => {
+      const contributionId = await contribution.contributionsCount();
+      return assertRevert(async () => {
+        await contribution.veto(contributionId.toNumber(), {from: member1});
+        'sender does not have permission to veto';
+      });
+    });
+
+    it("should revert when veto contribution that does not exist", async () => {
+      const contributionId = await contribution.contributionsCount();
+      return assertRevert(async () => {
+        await contribution.veto(contributionId.toNumber()+1, {from: root});
+        'contribution not found';
+      });
+    });
+
+    it("veto contribution", async () => {
+      const contributionId = await contribution.contributionsCount();
+      let contributionObject = await contribution.getContribution(contributionId.toNumber());
+      console.log("veto block: " + contributionObject[7]);
+      console.log("current block: " + await getBlockNumber());
+
+      await contribution.veto(contributionId.toNumber(), {from: root});
+      // eslint-disable-next-line no-undef
+      assert(contributionObject[9], true);
+    });
+  });
+
+  describe("Claim contribution", async () => {
+    let contributionId;
+
+    // eslint-disable-next-line no-undef
+    before(async () =>{
+      let amount = 200;
+      let contributorId = 1;
+      let hashDigest = '0x0000000000000000000000000000000000000000000000000000000000000000';
+      let hashFunction = 1;
+      let hashSize = 1;
+  
+      await contribution.add(amount, contributorId, hashDigest, hashFunction, hashSize, {from: root});
+    
+      contributionId = await contribution.contributionsCount();
+    });
+
+    it("should revert when claim contribution that does not exist", async () => {
+      return assertRevert(async () => {
+        await contribution.claim(contributionId.toNumber()+1, {from: root});
+        'contribution not found';
+      });
+    });
+
+    it("should revert when claim contribution before confirmation block", async () => {
+      return assertRevert(async () => {
+        await contribution.claim(contributionId.toNumber(), {from: root});
+        'contribution not confirmed yet';
+      });
+    });
+
+    it("claim contribution", async () => {
+      if(contributionId < 10) {
+        await timeTravel(100);
+      }
+      else {
+        await timeTravel(blocksToWait);
+      }
+      await mineBlock();
+
```

Comment:
can't figure out why this one is not working, idk what I'm missing there! @bumi @skddc